Source/VCExpr/TypeErasureArguments.cs

```diff
@@ -193,41 +193,23 @@ internal MapTypeAbstractionBuilderArguments(TypeAxiomBuilderArguments axBuilder,
         storeTypes[i] = AxBuilder.T;
       }
       // Fill in the map type
-      if (CommandLineOptions.Clo.MonomorphicArrays) {
-        selectTypes[i] = abstractedType;
-        storeTypes[i] = abstractedType;
-      } else {
-        selectTypes[i] = AxBuilder.U;
-        storeTypes[i] = AxBuilder.U;
-      }
+      selectTypes[i] = AxBuilder.U;
```

Comment:
I suppose this change is safe because we are not even constructing AxiomBuilder for monomorphic encoding?

Reply:
Yes. There is an invariant now that no AxiomBuilders get created/invoked for monomorphic programs.